misc/python/materialize/terminal.py

```diff
@@ -21,6 +22,8 @@
 STYLE_BOLD = "\033[1m"
 _END_FORMATTING = "\033[0m"
 
+_COLOR_FORMATTING_RE = re.compile(r"\[\d+m(.*?)\[0m")
```

Comment:
Hm, so the \033 is already somehow removed, probably because it's not printable. It's a bit unfortunate that you can't have a literal `[0m` in the text now. Maybe we should instead filter out the colors before the `\033` is filtered out (by junit-xml I guess?)

Reply:
I mean you could have a `[0m` literal but not when another formatting literal is before.

Anyway, you are absolutely right. The correct approach is to apply the transformation before creating the junit.xml file.